tools/cmake/tests/mbed_test_mode/main.cpp

```diff
@@ -0,0 +1,13 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2021 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if BUILD_TESTING && !defined(MBED_TEST_MODE)
```

Comment:
Should we put this test inside a `TARGET_IGNORE` folder to make Mbed CLI 1 ignore it? I'm not clear on how we avoid multiple definition of `main` errors without some way of excluding this test from normal builds of applications.

Maybe a path something like this?
```
tools/cmake/tests/TARGET_IGNORE/mbed_test_mode/main.cpp
```

Reply:
Isn't tools folder always ignored? If no, we will need unfortunately hide this. although `TESTS` should work as it does for our greentea tests?

Reply:
That might be the way this is excluded. If so, nothing to do here. I'd expect CI for examples to break if the hiding weren't working.